src/Support/SoapClients/DecoratedClient.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RicorocksDigitalAgency\Soap\Support\SoapClients;
+
+use RicorocksDigitalAgency\Soap\Contracts\PhpSoap\Client;
+use SoapClient;
+use SoapHeader;
+
+/**
+ * @internal
+ */
+final class DecoratedClient implements Client
+{
+    private SoapClient $client;
+
+    private string $endpoint = '';
+
+    /**
+     * @var array<string, mixed>
+     */
+    private array $options = [];
+
+    /**
+     * @param array<int, SoapHeader> $headers
+     */
+    public function setHeaders(array $headers): static
+    {
+        $this->client()->__setSoapHeaders($headers);
+
+        return $this;
+    }
+
+    public function setEndpoint(string $endpoint): static
+    {
+        $this->endpoint = $endpoint;
+
+        return $this;
+    }
+
+    /**
+     * @param array<string, mixed> $options
+     */
+    public function setOptions(array $options): static
+    {
+        $this->options = $options;
+
+        return $this;
+    }
+
+    public function call(string $method, mixed $body): mixed
+    {
+        return $this->client()->{$method}($body);
+    }
+
+    public function getFunctions(): array
+    {
+        return $this->client()->__getFunctions();
+    }
+
+    public function __get(string $name): mixed
```

Comment:
what is this being used for exactly? There are no public properties on the client IIRC.